tools/pythonpkg/tests/fast/arrow/test_dataset.py

```diff
@@ -81,4 +81,53 @@ def test_parallel_dataset_roundtrip(self,duckdb_cursor):
         # turn it into an arrow table
         arrow_table_2 = pyarrow.Table.from_pandas(df)
 
-        assert arrow_table.equals(arrow_table_2)
\ No newline at end of file
+        assert arrow_table.equals(arrow_table_2)
+
+    def test_ducktyping(self, duckdb_cursor):
+        duckdb_conn = duckdb.connect()
+        dataset = CustomDataset()
+        query = duckdb_conn.execute("SELECT b FROM dataset WHERE a < 5")
+        record_batch_reader = query.fetch_record_batch(2048)
+        arrow_table = record_batch_reader.read_all()
+        assert arrow_table.equals(DATA[:5].select(['b']))
+
+
+if can_run:
```

Comment:
While you're changing this file, can you update it to use pytests importorskip instead?